### PR TITLE
feat: Reimplement the graph code for C

### PR DESCRIPTION
**Pull request by krahets**
If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference code.
- [x] I've checked the code (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

**Review by Gonglja — COMMENTED** [review]
Since I'm not very familiar with this part of the algorithm, I can only review it syntactically, and the algorithm will need to be reviewed by others

File: codes/c/chapter_graph/graph_adjacency_matrix.c [hunk 1]
```diff
@@ -6,211 +6,140 @@
...
-            memcpy(graph->adjMat[i], graph->adjMat[i + 1], sizeof(int) * graph->size); // 被删除行的下方所有行上移
-            for (int j = index; j < graph->size; j++) {
-                graph->adjMat[i][j] = graph->adjMat[i][j + 1]; // 被删除列后的所有列前移
-            }
+    // 在邻接矩阵中删除索引 index 的行
+    for (int i = index; i < graph->size - 1; i++) {
+        for (int j = 0; j < graph->size; j++) {
+            graph->adjMat[i][j] = graph->adjMat[i + 1][j];
+        }
+    }
```
[the remark this answers is not in the document]
Comment (krahets) [reply codes/c/chapter_graph/graph_adjacency_matrix.c:61]: We do it in `addVertex()`

File: codes/c/chapter_graph/graph_dfs.c [hunk 2]
```diff
@@ -6,105 +6,69 @@
...
 
 /* Driver Code */
 int main() {
-    GraphAdjList *graph = newGraphAdjList(10);
-    for (int i = 0; i < 7; i++) {
-        addVertex(graph, i);
+    // 初始化无向图
+    int vals[] = {0, 1, 2, 3, 4, 5, 6};
+    int size = sizeof(vals) / sizeof(vals[0]);
+    Vertex **v = valsToVets(vals, size);
```
[the remark this answers is not in the document]
Comment (krahets) [reply codes/c/chapter_graph/graph_dfs.c:50]: Solved

**Comment by krahets** [comment]
Thanks for the review @Gonglja !

**Pull request merged**